service/history/historyEngine.go

```diff
@@ -1492,8 +1493,13 @@ func (e *historyEngineImpl) RespondActivityTaskCompleted(
 				// Unable to add ActivityTaskCompleted event to history
 				return &workflow.InternalServiceError{Message: "Unable to add ActivityTaskCompleted event to history."}
 			}
+			activityStartedTime = ai.StartedTime
 			return nil
 		})
+	if err == nil && !activityStartedTime.IsZero() {
+		e.metricsClient.RecordTimer(metrics.HistoryRespondActivityTaskCompletedScope, metrics.ActivityE2ELatency, time.Since(activityStartedTime))
```

Comment:
Pending review, let me try also add activity and workflow type

Reply:
and task list.

Reply:
resolved